optimizely/decision/feature_experiment_service.go

```diff
@@ -18,20 +18,53 @@
 package decision
 
 import (
+	"fmt"
+
 	"github.com/optimizely/go-sdk/optimizely/entities"
 )
 
 // FeatureExperimentService helps evaluate feature test associated with the feature
 type FeatureExperimentService struct {
-	featureExperimentService ExperimentService
+	compositeExperimentService ExperimentService
 }
 
 // NewFeatureExperimentService returns a new instance of the FeatureExperimentService
-func NewFeatureExperimentService() *CompositeExperimentService {
-	return NewCompositeExperimentService()
+func NewFeatureExperimentService() *FeatureExperimentService {
+	return &FeatureExperimentService{
+		compositeExperimentService: NewCompositeExperimentService(),
+	}
 }
 
 // GetDecision returns a decision for the given feature test and user context
-func (f FeatureExperimentService) GetDecision(decisionContext ExperimentDecisionContext, userContext entities.UserContext) (ExperimentDecision, error) {
-	return f.featureExperimentService.GetDecision(decisionContext, userContext)
+func (f FeatureExperimentService) GetDecision(decisionContext FeatureDecisionContext, userContext entities.UserContext) (FeatureDecision, error) {
+	feature := decisionContext.Feature
+	// @TODO this can be improved by getting group ID first and determining experiment and then bucketing in experiment
+	for _, experiment := range feature.FeatureExperiments {
```

Comment:
Nit: In other SDKs do we just look at the first feature experiment and ignore the rest?

Reply:
That's because we also look at the GroupID of the feature to perform the mutex. That enhancement needs to be made here first before we can stop at the first